pkg/parse/listener_impl.go

```diff
@@ -377,30 +377,26 @@ func (s *TreeShapeListener) ExitField_type(ctx *parser.Field_typeContext) {
 func makeTypeConstraint(t sysl.Type_Primitive, size_spec *parser.Size_specContext) []*sysl.Type_Constraint {
 	c := []*sysl.Type_Constraint{}
 	var err error
-	var l int
+	var l int64
 
 	switch t {
 	case sysl.Type_DATE, sysl.Type_DATETIME, sysl.Type_INT, sysl.Type_STRING:
 		val1 := size_spec.DIGITS(0).GetText()
-		if l, err = strconv.Atoi(val1); err == nil {
+		if l, err = strconv.ParseInt(val1, 10, 0); err == nil {
 			c = append(c, &sysl.Type_Constraint{
-				Length: &sysl.Type_Constraint_Length{
-					Max: int64(l),
-				},
+				Length: &sysl.Type_Constraint_Length{Max: l},
 			})
 		}
 	case sysl.Type_DECIMAL:
 		val1 := size_spec.DIGITS(0).GetText()
-		if l, err = strconv.Atoi(val1); err == nil {
+		if l, err = strconv.ParseInt(val1, 10, 0); err == nil {
 			c = append(c, &sysl.Type_Constraint{
-				Length: &sysl.Type_Constraint_Length{
-					Max: int64(l),
-				},
+				Length: &sysl.Type_Constraint_Length{Max: l},
 			})
 			if size_spec.DIGITS(1) != nil {
 				c[0].Precision = int32(l)
 				val1 = size_spec.DIGITS(1).GetText()
-				if l, err = strconv.Atoi(val1); err == nil {
+				if l, err = strconv.ParseInt(val1, 10, 0); err == nil {
```

Comment:
Approach was from https://github.com/securego/gosec/pull/422

`Atoi` docs say it's equivalent to `ParseInt(val, 10, 0)`, but `Atoi` returns an int(32) and `ParseInt` returns an `int64`. If we think the string really could be an `int64`, we shouldn't be parsing it into an `int`.

Reply:
Ahhh, I didn't notice the difference in function signature, since the docs for Atoi say "Atoi is equivalent to ParseInt(s, 10, 0)"
.... "converted to type int."
And I didn't read the second bit. Awesome, looks great!